saw-remote-api/saw-remote-api/Main.hs

```diff
@@ -20,32 +20,81 @@ import SAWServer.SetOption
 
 main :: IO ()
 main =
-  do theApp <- mkApp initialState sawMethods
+  do theApp <- mkApp "SAW RPC Server" serverDocs initialState sawMethods
      defaultMain description theApp
 
+serverDocs :: [Doc.Block]
+serverDocs =
+  [ Doc.Paragraph [Doc.Text "A remote server for SAW."]
+  ]
+
 description :: String
 description =
   "An RPC server for SAW."
 
-sawMethods :: [(Text, MethodType, JSON.Value -> Method SAWState JSON.Value)]
+sawMethods :: [AppMethod SAWState]
 sawMethods =
   -- Cryptol
-  [ ("SAW/Cryptol/load module",  Command, method cryptolLoadModule)
-  , ("SAW/Cryptol/load file",    Command, method cryptolLoadFile)
-  , ("SAW/Cryptol/save term",    Command, method saveTerm)
+  [ method
+     "SAW/Cryptol/load module"
+     Command
+     (Doc.Paragraph [Doc.Text "Load the specified Cryptol module."])
+     cryptolLoadModule
+  , method
+     "SAW/Cryptol/load file"
+     Command
+     (Doc.Paragraph [Doc.Text "Load the given file as a Cryptol module."])
+     cryptolLoadFile
+  , method
+     "SAW/Cryptol/save term"
+     Command
+     (Doc.Paragraph [Doc.Text "Save a term to be referenced later by name."])
+     saveTerm
   -- JVM
   {-
-  , ("SAW/JVM/load class",       Command, method jvmLoadClass)
-  , ("SAW/JVM/verify",           Command, method jvmVerify)
-  , ("SAW/JVM/assume",           Command, method jvmAssume)
+  , method "SAW/JVM/load class" Command (Doc.Paragraph [Doc.Text "TODO"]) jvmLoadClass
+  , method "SAW/JVM/verify"     Command (Doc.Paragraph [Doc.Text "TODO"]) jvmVerify
+  , method "SAW/JVM/assume"     Command (Doc.Paragraph [Doc.Text "TODO"]) jvmAssume
   -}
   -- LLVM
-  , ("SAW/LLVM/load module",     Command, method llvmLoadModule)
-  , ("SAW/LLVM/verify",          Command, method llvmVerify)
-  , ("SAW/LLVM/verify x86",      Command, method llvmVerifyX86)
-  , ("SAW/LLVM/assume",          Command, method llvmAssume)
+  , method
+     "SAW/LLVM/load module"
+     Command
+     (Doc.Paragraph [Doc.Text "Load the specified LLVM module."])
+     llvmLoadModule
+  , method
+     "SAW/LLVM/verify"
+     Command
+     (Doc.Paragraph [Doc.Text "Verify the named LLVM function meets its specification."])
+     llvmVerify
+  , method
+     "SAW/LLVM/verify x86"
+     Command
+     (Doc.Paragraph [Doc.Text $ T.pack
+                              $ "Verify an x86 function from an ELF file for use as "
+                                ++ "an override in an LLVM verification meets its specification."])
```

Comment:
It would be nice if there were a cleaner way to write long descriptions, but maybe that can wait for a future PR.